libs/isograph-react/src/iso.tsx

```diff
@@ -1,8 +1,52 @@
-export function iso<TResolverParameter>(
-  _queryText: TemplateStringsArray,
-): <TResolverReturn>(
-  x: (param: TResolverParameter) => TResolverReturn,
-) => (param: TResolverParameter) => TResolverReturn {
+type IdentityWithParam<TParam> = <TResolverReturn>(
+  x: (param: TParam) => TResolverReturn,
+) => (param: TParam) => TResolverReturn;
+
+type RepositoryLinkParam = {
+  data: {
+    id: string;
+    name: string;
+    owner: {
+      login: string;
+    };
+  };
+  [index: string]: any;
+};
+
+export type RepositoryPageType = {
+  data: {
+    // MEH
+    Header: any;
+    // MEH times two
+    RepositoryDetail: any;
+  };
+  [index: string]: any;
+};
+
+type UserLinkParams = {
+  data: {
+    login: string;
+  };
+  [index: string]: any;
+};
+
+type WhitespaceCharacter = ' ' | '\n';
+type Whitespace<In> = In extends `${WhitespaceCharacter}${infer In}` ? Whitespace<In> : In;
+
+type MatchesWhitespaceAndString<TString extends string, T> =
+  Whitespace<T> extends `${TString}${string}` ? T : never;
+
+export function iso<T>(
+  param: T & MatchesWhitespaceAndString<'field Repository.RepositoryLink', T>,
```

Comment:
I didn't try allowing whitespace in this string, but we could also just enforce in the compiler that no whitespace is allowed between the type name and the dot and the field name, and only one whitespace between the field and typename.